traitsui/qt4/directory_editor.py

```diff
@@ -35,11 +35,11 @@ class SimpleEditor(SimpleFileEditor):
 
     def _create_file_dialog(self):
         """Creates the correct type of file dialog."""
-        dlg = QtGui.QFileDialog(self.control)
-        dlg.selectFile(self._file_name.text())
-        dlg.setFileMode(QtGui.QFileDialog.FileMode.Directory)
-        dlg.setOptions(QtGui.QFileDialog.Option.ShowDirsOnly)
 
+        dlg = DirectoryDialog(
+            parent=self.get_control_widget(),
+            default_path=self._file_name.text(),
+        )
         return dlg
```

Comment:
should we not return dlg.control here so the return value stays a QtGui.QFileDialog instance as opposed to now a pyface `DirectoryDialog` instance?

Reply:
ah never mind I see we call open in `show_file_dialog` so the change is accounted for